### ISSUE TITLE
Refine grammar statement / expression

### ISSUE THREAD

**Issue by racodond** [iwarapter/sonar-puppet#148]
Assignment in if statement condition should raise a parse error:

```
if $a = 1 {}
```

**Comment by iwarapter** [comment]
This is because i chained the assignment in with the rest of the [operator expressions](https://github.com/iwarapter/sonar-puppet/blob/master/puppet-squid/src/main/java/com/iadams/sonarqube/puppet/api/PuppetGrammar.java#L506), we should be able to pull this out fairly easily to better comply with the [puppet grammar](https://github.com/puppetlabs/puppet/blob/3.8.0/lib/puppet/parser/grammar.ra#L316)

I'll come round to this when the remaining parser/rule issues are fixed.
